src/extension/popup/popupWindow.re

```diff
@@ -21,107 +20,114 @@ let select name onChange language lang =>
     <option value="ML"> (ReactRe.stringToElement "ML") </option>
   </select>;
 
-module PopupWindow = {
-  include ReactRe.Component.Stateful;
-  let name = "PopupWindow";
-  type props = {
-    inText: string,
-    outText: string,
-    inLang: Protocol.language,
-    outLang: Protocol.language,
-    link: string,
-    onOpen: string => unit,
-    onInputChanged: inLang::Protocol.language? => outLang::Protocol.language? => string => unit
-  };
-  type state = {
-    copyConfirmation: option string,
-    mutable inputRef: option ReactRe.reactRef,
-    inLanguage: Protocol.language,
-    outLanguage: Protocol.language
-  };
-  /* Init */
-  let getInitialState _ => {
+type state = {
+  copyConfirmation: option string, 
+  inputRef: option ReasonReact.reactRef,
+  inLanguage: Protocol.language,
+  outLanguage: Protocol.language
+};
+
+let showCopyConfirmation text () state self => {
+  Util.setTimeout (self.ReasonReact.update (fun () _ _ => ReasonReact.Update {...state, copyConfirmation: None})) 2500;
+  ReasonReact.Update {...state, copyConfirmation: Some text}
+};
+
+/*
+let updateInputRef nullableRef state _ =>
+  switch (Js.Null.to_opt nullableRef) {
+  | Some ref => ReasonReact.SilentUpdate {...state, inputRef: Some ref};
+  | None => ReasonReact.NoUpdate
+};
```

Comment:
It doesn't accept it as a nullable for some reason, which is really weird because that's clearly what `ReasonReact.element` actually requires: https://github.com/reasonml/reason-react/blob/master/src/reasonReact.rei#L155

cc @chenglou